src/components/molecules/Visualizer/hooks.ts

```diff
@@ -103,6 +104,50 @@ export default ({
   );
   dropRef(wrapperRef);
 
+  const viewportMobileMaxWidth = 768;
+  const defaultViewport = useMemo(
+    () => ({
+      width: 1080,
+      height: 720,
```

Comment:
Use `useRef<HTMLDivElement>` and measure width and height on initialization, or use 0. 1080 and 720 is not good because it is confusing that users cannot recognize if they are dummy value or not.